src/test/java/com/google/cloud/spanner/pgadapter/python/ITPsycopg2Test.java

```diff
@@ -367,4 +367,298 @@ public void testDeleteParameterized() throws IOException, InterruptedException {
+            + "False, "
+            + "74.567, "
+            + "69, "
+            + "Decimal('99.9999'), "
+            + "datetime.datetime(2021, 2, 16, 13, 18, 2, tzinfo=datetime.timezone.utc), "
+            + "datetime.date(2021, 7, 1), "
+            + "'bye bye world')"
+            + "\n";
+
+    actualOutput = executeWithoutParameters(testEnv.getPGAdapterPort(), sql2, "query");
+    assertEquals(expectedOutput, actualOutput);
+    // System.out.println(actualOutput);
+
+  }
+
+  @Test
+  public void testInsertUsingExecuteBatch() throws IOException, InterruptedException {
+    String sql =
+        "Insert into all_types("
+            + "col_bigint, "
+            + "col_bool , "
+            + "col_bytea , "
+            + "col_float8 , "
+            + "col_int , "
+            + "col_numeric , "
+            + "col_timestamptz , "
+            + "col_date , "
+            + "col_varchar )"
+            + " values("
+            + "%(bigint)s, "
+            + "%(bool)s, "
+            + "%(bytea)s, "
+            + "%(float)s, "
+            + "%(int)s, "
+            + "%(numeric)s, "
+            + "%(tstz)s, "
+            + "%(date)s, "
+            + "%(varchar)s)";
+
+    ArrayList<String> parameters = new ArrayList<>();
+    parameters.add("9");
+    parameters.add("bool");
+    parameters.add("True");
+    parameters.add("float");
+    parameters.add("89.23");
+    parameters.add("numeric");
+    parameters.add("33.546");
+    parameters.add("tstz");
+    parameters.add("2022-02-16T14:18:02+01:00");
+    parameters.add("varchar");
+    parameters.add("hello world");
+    parameters.add("date");
+    parameters.add("2022-7-1");
+    parameters.add("bigint");
+    parameters.add("11");
+    parameters.add("bytea");
+    parameters.add("alpha beta");
+    parameters.add("int");
+    parameters.add("34");
+
+    parameters.add("bool");
+    parameters.add("False");
+    parameters.add("float");
+    parameters.add("74.567");
+    parameters.add("numeric");
+    parameters.add("99.9999");
+    parameters.add("tstz");
+    parameters.add("2021-02-16T14:18:02+01:00");
+    parameters.add("varchar");
+    parameters.add("bye bye world");
+    parameters.add("date");
+    parameters.add("2021-7-1");
+    parameters.add("bigint");
+    parameters.add("16");
+    parameters.add("bytea");
+    parameters.add("gamma theta");
+    parameters.add("int");
+    parameters.add("69");
+
+    String actualOutput =
+        executeInBatch(testEnv.getPGAdapterPort(), sql, "named_execute_batch", parameters);
+    String expectedOutput = "1\n";
+
+    assertEquals(expectedOutput, actualOutput);
+
+    String sql2 =
+        "Select "
+            + "col_bigint, "
+            + "col_bool , "
+            + "col_float8 , "
+            + "col_int , "
+            + "col_numeric , "
+            + "col_timestamptz , "
+            + "col_date , "
+            + "col_varchar "
+            + "from all_types where col_bigint = 11 or col_bigint = 16";
```

Comment:
nit: here also add `order by id` and rename the `sql2` variable

Reply:
done